src/device/socket/vsock.rs

```diff
@@ -454,8 +444,15 @@ impl<H: Hal, T: Transport> VirtIOSocket<H, T> {
     }
 }
 
-fn read_header_and_body(buffer: &[u8], body: &mut [u8]) -> Result<VirtioVsockHdr> {
+fn read_header_and_body(
+    buffer: &[u8],
+    body: &mut [u8],
+    expected_packet_len: u32,
+) -> Result<VirtioVsockHdr> {
     let header = VirtioVsockHdr::read_from_prefix(buffer).ok_or(SocketError::BufferTooShort)?;
+    if expected_packet_len != header.packet_len()? {
```

Comment:
That doesn't look right. Shouldn't the VirtIO used buffer length match the total length of the data, including both the header and the body?

Reply:
Right. The packet_len() here is the total length including header and body.

Reply:
Ah right sorry, I misread.